tests/ceph_volume/test_ceph_volume_validate_persistent_naming.py

```diff
@@ -0,0 +1,106 @@
+import json
+from json import loads
+
+import yaml
+
+from cli.ceph.ceph import Ceph
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.utils import (
+    get_disk_devlinks,
+    get_lvm_on_osd_container,
+    get_running_containers,
+)
+
+
+class SpecFileCreationError(Exception):
+    pass
+
+
+class SpecFileApplyError(Exception):
+    pass
+
+
+def _identify_disks(node):
+    """
+    Identifies the disks associated with the node
+    Args:
+        node (ceph): Node in which the cmd is executed
+
+    Returns (list): List of available disks
+    """
+
+    disks = []
+    conf = {"format": "json"}
+    available_disks = json.loads(CephAdm(node).ceph.orch.device.ls(**conf))
+    for key in available_disks:
+        if key.get("devices"):
+            for dev in key.get("devices"):
+                disks.append(dev.get("path"))
+    return list(set(disks))
+
+
+def _create_spec_file(node, data_device, specs, host):
+    """
+    Creates the spec file as per the requirement
+    Args:
+        node (ceph): Node to execute cmd
+        data_device(str): data device details
+        specs(str): given spec
+        host(str): hostname of the node to add
+    """
+    path = "/root/osd.yaml"
+    specs["data_devices"]["paths"] = data_device
```

Comment:
Why these parameters need to set here ? Can't it be part of specs already ?

Reply:
These values are taken during the run time Vaibhav, so cannot be given via config.

Reply:
@pranavprakash20 `data_device` & `host` are passed to function which are only used to assign to `specs` in the function. Instead if set these value before a function call and pass `data_device`. `_create_spec_file` won't require variables `data_device` and `host`.

Reply:
Yes Vaibhav, makes sense. Missed that. Will revisit